pyro/infer/mcmc/util.py

```diff
@@ -208,3 +212,135 @@ def log_prob(self, model_trace):
         log_probs = self._get_log_factors(model_trace)
         with shared_intermediates() as cache:
             return contract_to_tensor(log_probs, self._enum_dims, cache=cache)
+
+
+def _guess_max_plate_nesting(model, args, kwargs):
+    """
+    Guesses max_plate_nesting by running the model once
+    without enumeration. This optimistically assumes static model
+    structure.
+    """
+    with poutine.block():
+        model_trace = poutine.trace(model).get_trace(*args, **kwargs)
+    sites = [site for site in model_trace.nodes.values()
+             if site["type"] == "sample"]
+
+    dims = [frame.dim
+            for site in sites
+            for frame in site["cond_indep_stack"]
+            if frame.vectorized]
+    max_plate_nesting = -min(dims) if dims else 0
+    return max_plate_nesting
+
+
+def _transform_fn(transforms, params, invert=True):
```

Comment:
Great to see this refactored out!

Reply:
Hmm, I thought that I have deleted this function because it is not used elsewhere. ^^!

Reply:
I meant `initialize_model` and its contents. I think this method can be deleted.